dash/App/Accounts/Add/Components/index.js

```diff
@@ -0,0 +1,278 @@
+import { useEffect, useRef, useState, useCallback } from 'react'
+
+import RingIcon from '../../../../../resources/Components/RingIcon'
+import { debounce } from '../../../../../resources/utils'
+import link from '../../../../../resources/link'
+import zxcvbn from 'zxcvbn'
+
+const navForward = async (newAccountType, accountData) =>
+  link.send('nav:forward', 'dash', {
+    view: 'accounts',
+    data: {
+      showAddAccounts: true,
+      newAccountType,
+      accountData
+    }
+  })
+
+const removeLineBreaks = (str) => str.replace(/(\r\n|\n|\r)/gm, '')
+
+function AddHotAccountWrapper({ children, title, svgName, summary, intro }) {
+  return (
+    <div className={'addAccountItem addAccountItemSmart addAccountItemAdding'}>
+      <div className='addAccountItemBar addAccountItemHot' />
+      <div className='addAccountItemWrap'>
+        <div className='addAccountItemTop'>
+          <div className='addAccountItemTopType'>
+            <div className='addAccountItemIcon'>
+              <div className='addAccountItemIconType addAccountItemIconHot'>
+                <RingIcon {...{ svgName }} />
+              </div>
+              <div className='addAccountItemIconHex addAccountItemIconHexHot' />
+            </div>
+            <div className='addAccountItemTopTitle'>{title}</div>
+          </div>
+          <div className='addAccountItemSummary'>{summary}</div>
+        </div>
+        <div className='addAccountItemOption'>
+          <div
+            className='addAccountItemOptionIntro'
+            onMouseDown={() => {
+              this.adding()
+              setTimeout(
+                () =>
+                  link.send('tray:action', 'navDash', {
+                    view: 'notify',
+                    data: { notify: 'hotAccountWarning', notifyData: {} }
+                  }),
+                800
+              )
+            }}
+          >
+            {intro}
+          </div>
+          <div className='addAccountItemOptionSetupFrames'>{children}</div>
+        </div>
+        <div className='addAccountItemFooter' />
+      </div>
+    </div>
+  )
+}
+
+function EnterSecret({ newAccountType, isValidSecret, title }) {
+  const [secret, setSecret] = useState('')
+  const [error, setError] = useState(null)
+
+  const updateInput = (e) => {
+    const value = removeLineBreaks(e.target.value)
+    setSecret(value)
+  }
+
+  const handleSubmit = (e) => {
+    if (e?.type !== 'mousedown' && e?.key !== 'Enter') {
+      return
+    }
+
+    if (!isValidSecret(secret)) {
+      return setError(`INVALID ${title.toUpperCase()}`)
+    }
+
+    return navForward(newAccountType, {
+      secret
+    })
+  }
+
+  return (
+    <div style={{ textAlign: 'center', width: '100%' }} className='addAccountItemOptionSetupFrame'>
+      <div data-testid='addHotAccountSecretTitle' className='addAccountItemOptionTitle'>
+        {title}
+      </div>
+      {error && (
+        <div data-testid='addHotAccountSecretError' style={{ color: 'red' }}>
```

Comment:
I noticed a bunch of these test ids. unless they're meant to be temporary, lets work together to see if we can structure the markup in a way that removes the need for them whenever possible.